api/src/main/java/io/opentelemetry/trace/AttributeValue.java

```diff
@@ -221,4 +320,92 @@ public final Type getType() {
     @Override
     public abstract double getDoubleValue();
   }
+
+  @Immutable
+  @AutoValue
+  abstract static class AttributeValueStringArray extends AttributeValue {
+
+    AttributeValueStringArray() {}
+
+    static AttributeValue create(String... stringValues) {
```

Comment:
Also, what should happen if `null` or empty strings are passed *inside* the array? I'm strongly against dropping them silently as done with #771 for single strings, as this would preclude use cases such as having corresponding key and value arrays where `value_array[i]` is the value for `key_array[i]`.

Reply:
The current implementation does not ignore null or empty strings *inside* the array and inserts them into the list of values for the given key.

Reply:
I'd be up with @Oberon00's suggestion on at least not allowing `null` values pass through (this check would also have to apply to the other array values).

Reply:
@carlosalberto I'm not sure that's the same thing I meant. What do you mean by not letting them pass through? I don't think that is necessarily the best idea but I'd still prefer it over dropping them. Maybe we should make them into empty strings?

Reply:
@Oberon00  Either way, I want us to be clear on the expectations, one way or another. I personally think it would be **very** dirty/lazy to let go through null values in the arrays - but if we decide another way, I'm fine as long as we document it.

Reply:
So what is the alternative if we don't let them go through, what do we do if we encounter one?

Reply:
I put this topic in the SIG notes for today, so we can have the feedback of the others

Reply:
@thisthat Maybe resolve this comment? IIRC this was resolved in a previous SIG call.